app/controllers/campaigns_controller.rb

```diff
@@ -87,9 +87,9 @@ def update
   end
 
   def destroy
-    @campaign.destroy
+    @campaign.update(status: :archived)
```

Comment:
I like how readable this is, but wonder if we should use the enum. Also, at some point we need to rethink/refactor how enums are managed in the app since they've forked far beyond a simple key/value structure. We don't have to address on this PR, just noting that we need to refactor it sometime.